src/main/resources/com/google/api/codegen/csharp/gapic_resourcenames.snip

```diff
@@ -409,6 +409,11 @@
             @join field : proto.fields
                 /// <summary>
                 /// A <see cref="{@field.docTypeName}"/>-typed view over the <see cref="{@field.underlyingPropertyName}"/> resource name property.
+                @if field.isOptional
```

Comment:
Hmm. I'm not sure I'd do it in the proto itself. That suggests that it's *always* either optional or required - whereas in reality it may be required in some cases, and optional in others.

I'd make the proto always accept null, setting the value to "". That's simple and dumb.

Then the *method* calls can validate that required parameters are non-null - assuming it already does so for required parameters of other types. (If it doesn't, let's not *just* do it for that.)